core/common/src/main/scala/org/typelevel/otel4s/context/propagation/TextMapPropagator.scala

```diff
@@ -84,31 +74,70 @@ trait TextMapPropagator[F[_], Ctx] {
     * @return
     *   a copy of the carrier, with new fields injected
     */
-  def injected[A: TextMapUpdater](ctx: Ctx, carrier: A): A
+  def inject[A: TextMapUpdater](ctx: Ctx, carrier: A): A
 }
 
 object TextMapPropagator {
 
-  def apply[F[_], Ctx](implicit
-      ev: TextMapPropagator[F, Ctx]
-  ): TextMapPropagator[F, Ctx] = ev
+  /** Creates a [[TextMapPropagator]] which delegates injection and extraction
+    * to the provided propagators.
+    *
+    * @example
+    *   {{{
+    * val w3cPropagator: TextMapPropagator[Context] = ???
+    * val httpTracePropagator: TextMapPropagator[Context] = ???
+    * val textMapPropagator = TextMapPropagator.composite(w3cPropagator, httpTracePropagator)
+    *   }}}
+    * @tparam Ctx
+    *   the context to use to extra or inject data
+    */
+  def composite[Ctx](
+      propagators: TextMapPropagator[Ctx]*
+  ): TextMapPropagator[Ctx] =
+    propagators.toList match {
+      case Nil          => new Noop[Ctx]
+      case head :: Nil  => head
+      case head :: tail => new Multi(NonEmptyList(head, tail))
```

Comment:
would it be a good idea in this case to preemptively flatten out any nested `Multi`s? otherwise you could theoretically overflow the stack on `#fields`.

also, with `Multi` being private, I don't think it really needs specifically `NonEmptyList`, if working around that for the above concern becomes annoying

Reply:
A also made a `Monoid` instance that covers this particular case:

```scala
implicit def textMapPropagatorMonoid[Ctx]: Monoid[TextMapPropagator[Ctx]] =
  new Monoid[TextMapPropagator[Ctx]] {
    val empty: TextMapPropagator[Ctx] = noop[Ctx]
    def combine(
        x: TextMapPropagator[Ctx],
        y: TextMapPropagator[Ctx]
    ): TextMapPropagator[Ctx] =
      (x, y) match {
        case (_: Noop[Ctx], _: Noop[Ctx]) =>
          empty
        case (that, _: Noop[Ctx]) =>
          that
        case (_: Noop[Ctx], other) =>
          other
        case (that: Multi[Ctx], other: Multi[Ctx]) =>
          new Multi[Ctx](that.propagators ++ other.propagators)
        case (that: Multi[Ctx], other) =>
          new Multi[Ctx](that.propagators :+ other)
        case (that, other: Multi[Ctx]) =>
          new Multi[Ctx](that +: other.propagators)
        case (that, other) =>
          new Multi[Ctx](List(that, other))
      }
  }
  ```
  
  Should I add it too?

Reply:
go for it! although I think the first case is redundant given the following two

Reply:
I replaced `NonEmptyList` with `List`, added `Monoid[TextMapPropagator[Ctx]], and added tests.